src/Components/Server/src/ComponentHub.cs

```diff
@@ -223,6 +224,17 @@ public async ValueTask ReceiveByteArray(int id, byte[] data)
             await circuitHost.ReceiveByteArray(id, data);
         }
 
+        public async ValueTask<bool> ReceiveJSDataChunk(long streamId, byte[] chunk, string error)
+        {
+            var circuitHost = await GetActiveCircuitAsync();
+            if (circuitHost == null)
+            {
+                return false;
+            }
+
+            return await circuitHost.ReceiveJSDataChunk(streamId, chunk, error);
```

Comment:
I don't think we need to await here. 

Doing so will prevent other messages from being processed until this one completes (I believe) and cause unnecessary contention.

We already do all exception handling inside CircuitHost and ensure the execution is serialized via dispatching within the sync context, so I believe this can be avoided.

Similar comment for ReceiveByteArray (which we might have missed on the previous PR). If you have any concerns, let's discuss the details.

/cc: @SteveSandersonMS

Reply:
The ReceiveByteArray was the subject of a [previous discussion](https://github.com/dotnet/aspnetcore/pull/33015#discussion_r639313770) as well. I think it should be fine to remove the await and stay in line with the other handlers.

Reply:
Yeah, I see now that we need to await this one

Reply:
Which potentially can mean we freeze if the pipe blocks?

Reply:
> Doing so will prevent other messages from being processed until this one completes (I believe)

Not sure about that. The sync context prevents multiple simultaneous executions, but doesn't prevent multiple interleaved async operations. As long as we yield the thread, we're allowing other messages to be processed in the meantime.

Or am I missing something?

Reply:
SignalR only processes one message at a time, and in this case we are waiting for the invocation to finish?

Reply:
> SignalR only processes one message at a time, and in this case we are waiting for the invocation to finish?

Right, yes, that's correct. Glad you noticed this.

So like all the other `ComponentHub` hub methods, we should fire-and-forget anything that takes an indeterminate amount of time.

Reply:
So we currently `await` as we expect a `bool` return value which represents the status of the circuit (ie. tells JS whether or not it should keep sending chunks). Given we can't await since SignalR only processes one message at a time, should we implement a `cancelStream` endpoint on the client to notify when .NET has disposed of the stream for whatever reason?

Reply:
Hey @BrennanConroy, could you please clarify what the max SignalR queuing for unprocessed messages would be? What happens if the limit is reached? Does it eventually throw / drop TCP packets?